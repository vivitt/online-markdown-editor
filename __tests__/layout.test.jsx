import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import Layout from "../app/layout";

describe("Layout", () => {
  it("renders a nav element", () => {
    render(
      <Layout>
        <div>Dummy</div>
      </Layout>
    );
    const nav = screen.getByRole("navigation");
    expect(nav).toBeInTheDocument();
  });

  it("renders homepage unchanged", () => {
    const { container } = render(
      <Layout>
        <div>Dummy</div>
      </Layout>
    );
    expect(container).toMatchSnapshot();
  });
});
