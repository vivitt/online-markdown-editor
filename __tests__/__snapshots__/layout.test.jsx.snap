// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders homepage unchanged 1`] = `
<div>
  <html
    lang="en"
  >
    <body
      class="className"
    >
      <nav>
        <ul
          class="flex"
        >
          <li>
            <a
              href="./"
            >
              Home
            </a>
          </li>
          <li>
            <a
              href="./editor"
            >
              Editor
            </a>
          </li>
        </ul>
      </nav>
      <div>
        Dummy
      </div>
    </body>
  </html>
</div>
`;
